UnchainedLauncherCore/src/API/ServerBrowser.cs

```diff
@@ -0,0 +1,248 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Net;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using UnchainedLauncher.Core.API;
+using UnchainedLauncher.Core.JsonModels.Metadata.V3;
+using System.Runtime.Loader;
+using PropertyChanged;
+
+namespace UnchainedLauncher.Core.API
+{
+    public record Ports(
+        int Game,
+        int Ping,
+        int A2s
+    );
+    [AddINotifyPropertyChangedInterface]
+    public record C2ServerInfo {
+        public bool PasswordProtected { get; set; } = false;
+        public string Name { get; set; } = "";
+        public string Description { get; set; } = "";
+        public Ports Ports { get; set; } = new(7777, 3075, 7071);
+        // TODO: The selection of the Mod datatype here is potentially
+        // incorrect. Change it to whatever is easier and works with the backend
+        public Mod[] Mods { get; set; } = Array.Empty<Mod>(); 
```

Comment:
I really ought to update the openapi spec on the server backend implementation, then we could just generate models from that